Modules/vc-module-export/VirtoCommerce.ExportModule.Data/Extensions/ExportedTypeMetadataExtensions.cs

```diff
@@ -0,0 +1,119 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using VirtoCommerce.ExportModule.Core.Model;
+using VirtoCommerce.Platform.Core.Common;
+
+namespace VirtoCommerce.ExportModule.Data.Extensions
+{
+    public static class ExportedTypeMetadataExtensions
+    {
+        private class ExportTypePropertyInfoEx
+        {
+            public ExportedTypeColumnInfo ColumnInfo { get; set; }
+            public bool IsReference { get; set; }
+            public PropertyInfo PropertyInfo { get; set; }
+        }
+
+        /// <summary>
+        /// Returns basic info about potentially exported fields of entity.
+        /// </summary>
+        /// <typeparam name="T">Type for getting metadata.</typeparam>
+        /// <param name="extractReferenceProperties">If true, reference properties properties (<see cref="Enitity"/> and collection of <see cref="Enitity"/>) will be extracted recursively.</param>
+        /// <returns>Metadata for the T type.</returns>
+        public static ExportedTypeMetadata GetFromType(this Type type, bool extractReferenceProperties)
```

Comment:
Let's rename to `GetPropertyNames`, as we use this as extension

Reply:
OK